File: react-client/src/services/StudentService.js
import axios from "axios";

const STUDENTS_REST_API_URL = 'http://localhost:8083/api/students';

class StudentService {
    getStudents() {
        return axios.get(STUDENTS_REST_API_URL);
    }
    createStudent(student){
        return axios.post(STUDENTS_REST_API_URL, student)
    }
    updateStudent(student, studentId){
        return axios.put(STUDENTS_REST_API_URL + '/' + studentId, student);
    }
    deleteStudent(studentId){
        return axios.delete(STUDENTS_REST_API_URL + '/' + studentId);
    }
}

export default new StudentService()

File: Spring-server/src/main/java/com/service/StudentService.java
package com.service;

import com.model.Student;

import java.util.List;

public interface StudentService {
    public List<Student> getStudents();
    public Student createStudent(Student student);
    public Student getStudentById(Long id);
    public Student updateStudent(Long id,Student studentDetails);
    public void deleteStudent(Long id);
    public boolean isStudentExist(String name);
}
